index.js

```diff
@@ -1,27 +1,73 @@
-const mongoose = require('mongoose');
+const mongoose = require("mongoose");
+const Recipe = require("./models/Recipe.model");
+const data = require("./data");
 
-// Import of the model Recipe from './models/Recipe.model.js'
-const Recipe = require('./models/Recipe.model');
-// Import of the data from './data.json'
-const data = require('./data');
+const MONGODB_URI = "mongodb://localhost:27017/recipe-app";
 
-const MONGODB_URI = 'mongodb://localhost:27017/recipe-app';
+mongoose.set("useFindAndModify", false);
+mongoose.set("debug", true);
 
-// Connection to the database "recipe-app"
 mongoose
   .connect(MONGODB_URI, {
     useCreateIndex: true,
     useNewUrlParser: true,
-    useUnifiedTopology: true
+    useUnifiedTopology: true,
   })
-  .then(self => {
+  .then((self) => {
     console.log(`Connected to the database: "${self.connection.name}"`);
-    // Before adding any documents to the database, let's delete all previous entries
     return self.connection.dropDatabase();
   })
   .then(() => {
-    // Run your code here, after you have insured that the connection was made
+    Recipe.create(recipeObj)
+      .then((results) => console.log(`Saved new recipe: ${results}`))
+      .catch((saveErr) => console.log(`Save failed: ${saveErr}`));
+
+    Recipe.insertMany(data).then((results) => {
+      results.forEach((result) => {
+        console.log(`Recipe titles: ${result.title}`);
+      });
+
+      Recipe.findOneAndUpdate(
+        { title: "Rigatoni alla Genovese" },
+        { $set: { duration: 100 } },
+        { new: true }
+      )
+        .then(() => console.log("Data modified succesfully"))
+        .catch((err) => console.error(`Error: ${err}`));
+
+      Recipe.deleteOne(
+        {
+          title: "Carrot Cake",
+        },
+        {
+          new: true,
+        }
+      )
+        .then(() => console.log("Data is deleted"))
+        .catch((err) => console.error(`Error: ${err}`));
+
+      mongoose.connection.close();
+    });
   })
```

Comment:
Hi Emma, 

So the problem with your solution on line 21-49 is that you have several promises that are executing(create, insertMany, update and delete) at the same time and because each runs asynchronously there's a high chance that you won't be able "update" or "delete" because "insertMany" isn't completely done for example.

To prevent that a possible solution would be to have each promise happen in separates `.then()` so that each action would have to be completed before moving on to the next one.

Here's an example of what I mean
```
.then(() => {
    // Run your code here, after you have insured that the connection was made
    Recipe.create(recipeObj)
      .then((results) => console.log(`Saved new recipe: ${results}`))
      .catch((saveErr) => console.log(`Save failed: ${saveErr}`));
})
  .then(recipe => {
    return Recipe.insertMany(data)
      .then((results) => console.log(`Saved new recipe: ${results}`))
      .catch((saveErr) => console.error(`Save failed: ${saveErr}`))
    console.log(recipe.title)
  })
  //ITERATION 4- Update recipe
  .then(recipe => {
    return Recipe.findOneAndUpdate({ title: "Rigatoni alla Genovese" }, { duration: 100 })
      .then(() => console.log(`Duration is updated`))
      .catch((saveErr) => console.error(`Save failed: ${saveErr}`));
  })
  .catch(error => {
    console.error('Error connecting to the database', error);
  });

//ITERATION 5- Remove a recipe
  .then(recipe => {
    return Recipe.deleteOne({ title: "Carrot Cake" })
    .then(() => console.log(`Carrot Cake is deleted`))
    .catch((saveErr) => console.error(`Save failed: ${saveErr}`))
  })
//ITERATION 6- Close the Database
  .then(recipe => {
  return mongoose.connection.close()
      .then(() => console.log(`Database is closed!`))
      .catch(() => console.error('Delete Failed'))
    })
    .catch (error => {
      console.error('Error connecting to the database', error)
    })